spec/models/footer_spec.rb

```diff
@@ -0,0 +1,32 @@
+require "rails_helper"
+
+RSpec.describe Footer, type: :model do
+  before {
+    allow(Rails).to receive(:root).and_return(file_fixture("_settings").join("..").cleanpath)
+  }
+
+  describe "#links" do
+    it "reads urls correctly" do
+      footer = described_class.load file_fixture("_settings/footer.yml").cleanpath
```

Comment:
[suggestion, non-blocking] an rspec convention is that the object under test is called `subject`. It can be set by adding

```
subject { described_class.load file_fixture("_settings/footer.yml").cleanpath }
```

Inside either `describe` or `context` blocks (in this file, I'd put it where you have the `before`). Then you don't need to re-define it in each test.

All test files automatically have a `subject`, defined as `subject { described_class.new }` if you don't set a custom one.

Reply:
🤯 Well that's nice!